### PR TITLE
feat: computed members map

### PR DESCRIPTION
**Pull request by zied-elouaer**

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -67,6 +67,17 @@ variable "policy_bindings" {
   default     = null
 }
 
+variable "computed_members_map" {
+  type        = map(string)
+  description = "(Optional) A map of members to replace in 'members' to handle terraform computed values. Will be ignored when policy bindings are used."
+  default     = {}
+
+  validation {
+    condition     = alltrue([for k, v in var.computed_members_map : can(regex("^(user|serviceAccount|group|domain):", v))])
```
Comment (waxb) [inline variables.tf:76]: `members` can have [these](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/google_service_account_iam#member/members) values, validation is too strict

**Review by waxb — CHANGES_REQUESTED** [review]

**Review by mariux — CHANGES_REQUESTED** [review]

File: variables.tf [hunk 2]
```diff
@@ -67,6 +67,17 @@ variable "policy_bindings" {
   default     = null
 }
 
+variable "computed_members_map" {
+  type        = map(string)
+  description = "(Optional) A map of members to replace in 'members' to handle terraform computed values. Will be ignored when policy bindings are used."
+  default     = {}
+
+  validation {
+    condition     = alltrue([for k, v in var.computed_members_map : can(regex("^(allUsers|allAuthenticatedUsers|(user|serviceAccount|group|domain):)", v))])
```
Comment (waxb) [inline variables.tf:76]: `principalSet` is also an acceptable string here

**Review by waxb — CHANGES_REQUESTED** [review]

File: variables.tf [hunk 3]
```diff
@@ -67,6 +67,17 @@ variable "policy_bindings" {
...
 }
 
+variable "computed_members_map" {
+  type        = map(string)
+  description = "(Optional) A map of members to replace in 'members' to handle terraform computed values. Will be ignored when policy bindings are used."
+  default     = {}
+
+  validation {
+    condition     = alltrue([for k, v in var.computed_members_map : can(regex("^(allUsers|allAuthenticatedUsers|(user|serviceAccount|group|domain|principalSet):)", v))])
+    error_message = "The value must be a non-empty string being a valid principal type identified with `allUsers`, `allAuthenticatedUsers` or prefixed with `user:`, `serviceAccount:`, `group:`, `domain:`, or `principalSet:`."
```
Comment (waxb) [inline variables.tf:77]:
```suggestion
    error_message = "The value must be a non-empty string being a valid principal type identified with `allUsers`, `allAuthenticatedUsers` or prefixed with `user:`, `serviceAccount:`, `group:`, `domain:` or `principalSet:`."
```

Comment (zied-elouaer) [reply variables.tf:77]: https://www.grammarly.com/blog/what-is-the-oxford-comma-and-why-do-people-care-so-much-about-it/

**Review by mariux — APPROVED** [review]
lgtm

**Pull request merged**